src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpDoNotUseStackallocInLoops.cs

```diff
@@ -23,6 +23,12 @@ public sealed override void Initialize(AnalysisContext context)
             context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
             context.RegisterSyntaxNodeAction(ctx =>
             {
+                // stackalloc is in the for loop initializer and thus not part of the actual loop.
+                if (ctx.Node.Parent is EqualsValueClauseSyntax { Parent: VariableDeclaratorSyntax { Parent: VariableDeclarationSyntax { Parent: ForStatementSyntax } } })
+                {
+                    return;
+                }
+
```

Comment:
Won't this result in false negatives if this loop is nested in another?

I'd expect we'd instead want this at another special-case like https://github.com/dotnet/roslyn-analyzers/pull/6728/files#diff-b65a77da1c4644408673499e08231ec7bed47aa0a42e1b7097911f74ba19cfe1L38

Reply:
Yes, you're right. I've added it as a special case, just like `foreach` and added another test for it.